src/cell.cpp

```diff
@@ -922,6 +922,39 @@ openmc_cell_set_temperature(int32_t index, double T, const int32_t* instance)
   return 0;
 }
 
+extern "C" int
+openmc_cell_get_temperature(int32_t index, double* T, const int32_t* instance)
+{
+  if (index < 0 || index >= model::cells.size()) {
+    strcpy(openmc_err_msg, "Index in cells array is out of bounds.");
+    return OPENMC_E_OUT_OF_BOUNDS;
+  }
+
+  Cell& c {*model::cells[index]};
+
+  if (c.sqrtkT_.size() < 1) {
+    strcpy(openmc_err_msg, "Cell temperature has not yet been set.");
+    return OPENMC_E_UNASSIGNED;
+  }
+
+  if (instance) {
+    if (*instance >= 0 && *instance < c.sqrtkT_.size()) {
+      *T = c.sqrtkT_[*instance] * c.sqrtkT_[*instance] / K_BOLTZMANN;
```

Comment:
Along the lines of [this morning's conversation](https://github.com/openmc-dev/openmc/pull/1224#issuecomment-488260908), if the cell instances are sharing a common temperature we'll want to return the first (and only) value in the vector as the cell temperature.

```cpp
if (*instance >= 0 && *instance < c.n_instances_) {
  double sqrtkT = c.sqrtkT_.size() == 1 ? c.sqrtkT_[0] : c.sqrtkT_[*instance];
  *T = sqrtkT * sqrtKT / K_BOLTZMAN;
} else { 
...
}
```

Reply:
Ah, okay, so multiple instances of a cell may exist if they either use unique temperatures + materials or they share a common temperature + material?

Reply:
(Sorry for the delay)

Yup yup. For clarity, it is possible for cell instances can have either unique temperatures or unique materials or both, depending on the input.